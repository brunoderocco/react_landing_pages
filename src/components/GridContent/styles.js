import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        text-align: center;
        max-width: 58rem;
        amrgin: 0 auto;
    `}
`;

export const Html = styled.div`
    ${({theme}) => css`
        margin: ${theme.spacings.xhuge};
    `}
`;