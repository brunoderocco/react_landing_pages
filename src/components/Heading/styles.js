import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkColor, size, uppercase }) => css`

    color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
