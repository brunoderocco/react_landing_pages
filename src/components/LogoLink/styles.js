import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
    font-size: ${theme.font.sizes.medium};
    font-weight: bolder;

    >img{
      max-height: 3rem;
    }
  `}
`;
