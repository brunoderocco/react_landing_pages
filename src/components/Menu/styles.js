import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top:0;
    left: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};
  `}
`;
