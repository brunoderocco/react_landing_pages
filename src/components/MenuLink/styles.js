import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css``}
`;
