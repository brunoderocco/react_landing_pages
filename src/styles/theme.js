export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#DC143C',
    white: 'white',
    lightBg: '#FFFFFF',
    darkBg: '#0A1128',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.2rem',
    },
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {},
};
