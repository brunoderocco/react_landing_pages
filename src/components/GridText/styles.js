import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextComponent} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;