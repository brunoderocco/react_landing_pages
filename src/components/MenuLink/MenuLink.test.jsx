import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { MenuLink } from '.';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link="http://localhost">Children</MenuLink>);
    expect(screen.getByRole('link')).toBeInTheDocument();
  });
});
