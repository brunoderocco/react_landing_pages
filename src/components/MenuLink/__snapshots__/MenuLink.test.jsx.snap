// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 2.4rem;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 35%;
  width: 25%;
  height: 0.2rem;
  background: #DC143C;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<div>
  <a
    class="c0"
    href="http://localhost"
    target="_self"
  >
    Children
  </a>
</div>
`;
